tests/fixtures.ts

```diff
@@ -0,0 +1,110 @@
+        const response = await fetch(
+            process.env.APPLICATION_BASE_URL + path, 
+            {
+                body: data,
+                headers: headers,
+                method: 'POST',
+            }
+        );
+        return response;
+    }
+
+    async get (path: string, headers: any): Promise<Response> {
+        const response = await fetch(
+            path, 
+            {
+                headers: headers,
+                method: 'GET',
+            }
+        );
+        return response;
+    }
+
+    async put (path: string, data: any, headers: any): Promise<Response> {
+        const response = await fetch(
+            process.env.APPLICATION_BASE_URL + path, 
+            {
+                body: JSON.stringify(data),
+                headers: headers,
+                method: 'PUT',
+            }
+        );
+        return response;
+    }
+
+    async patch(path: string, data: any, headers: any): Promise<Response> {
+        const response = await fetch(
+            process.env.APPLICATION_BASE_URL + path, 
+            {
+                body: JSON.stringify(data),
+                headers: headers,
+                method: 'PATCH',
+            }
+        );
+        return response;
+    }
+}
+
+export class DIDFixture extends HTTPHelper {
+    did: string;
+    didDocument: DIDDocument;
+
+    constructor() {
+        super();
+        this.did = "";
+        this.didDocument = {} as DIDDocument;
+    }
+    async createDid(network='testnet'): Promise<string> {
+        const response = await this.post(
+            `/did/create`, 
+            JSON.stringify({
+                    "methodSpecificIdAlgo": "uuid",
+                    "verificationMethodType": "Ed25519VerificationKey2018",
+                    "assertionMethod": true,
+                    "network": network,
+            }),
+            {
+                "Content-Type": "application/json",
+            }
+        );
+        if (!response.ok) {
+            console.error(await response.text());
+        }
+        console.error(await response.text());
+        const body = await response.json();
+        this.did = body.did;
+        return body.did;
+    }
+
+    async resolveDid(DIDString: string): Promise<DIDResolutionResult> {
+        const response = await this.get(`/did/search/${DIDString}`, {});
+        if (!response.ok) {
+            console.error(await response.text());
+        }
+        const body = await response.json();
+        this.didDocument = body.didDocument;
+        return body;
+    }
+}
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+export const test = base.extend<{}, WorkerScopedDIDFixture>({
+
+   DIDRepository: [
+    async ({ browser }, use) => {
+        expect(browser.isConnected()).toBeTruthy();
+        const didFixture = new DIDFixture();
+        await didFixture.createDid();
+        await use(didFixture);
+    }, 
+    { scope: 'worker'}]
+})
```

Comment:
I get the idea behind the DIDFixture a bit more, but I'm not sure if it's used in the intended way. A good example of a fixture would be for example collating repeatable elements and then using them in tests using extends. So a good example of a fixture would be:

1. In global setup, do stuff like login etc.
2. Create fixtures such as:
  1. You can successfully GET an account (this proves you're logged in)
  2. Account has balance
  3. A given DID exists
4. For DID tests, call the fixtures 1 and 2 above before attempting a DID create (since this is async, DID creation will not be kicked off until those conditions are satisfied.)
5. Call fixture 3 before attempting DID update
6. Call fixtures 1 and 3 before doing anything under Credentials API

You get the idea. It's a sanity check before you proceed with other test actions.

I wouldn't make DID _creation_ a fixture because it's a test on its own. _Checking_ it exists is a good use of a fixture though, because it ensures the test failure isn't because a race condition happened, and something like credential status create was attempted before the DID for it existed.

Reply:
I think that DID creation fixture could be useful at least for checking the general list of steps from user for issuing the credential. Like CreateDID->CreateStatusList->IssueCredential->Revoke/suspend. So for all of this steps we could use DID fixture which automatically checks the DID creation and could be reused for all of other checks.